cli/src/commands/dsl/validate.ts

```diff
@@ -97,63 +105,76 @@ export default class Validate extends BaseCommand {
 
     const allDiagnostics = ts.getPreEmitDiagnostics(program);
 
-    let diagnosticsCount = 0;
-
     const groupedDiagnostics = allDiagnostics.reduce(
-      (
-        acc: {
-          [key: string]: any[];
-        },
-        diagnostic
-      ) => {
+      (acc, diagnostic) => {
         const fileName = diagnostic.file?.fileName;
+        const category = diagnostic.category;
+
         if (fileName && files.includes(fileName)) {
-          if (!acc[fileName]) {
-            acc[fileName] = [];
+          const key =
+            category === ts.DiagnosticCategory.Error ? "errors" : "warnings";
+
+          if (!acc[key][fileName]) {
+            acc[key][fileName] = [];
           }
 
-          acc[fileName].push(diagnostic);
-          diagnosticsCount += 1;
+          acc[key][fileName].push(diagnostic);
         }
 
         return acc;
       },
-      {}
+      { errors: {}, warnings: {} } as {
+        errors: Record<string, ts.Diagnostic[]>;
+        warnings: Record<string, ts.Diagnostic[]>;
+      }
     );
 
-    const fileNameList = Object.keys(groupedDiagnostics);
-
-    fileNameList.forEach((diagnosticGroup) => {
-      this.log(`${diagnosticGroup}`);
-      groupedDiagnostics[diagnosticGroup].forEach((diagnostic) => {
-        if (diagnostic.file) {
-          const { line, character } = ts.getLineAndCharacterOfPosition(
-            diagnostic.file,
-            diagnostic.start!
-          );
-          const message = ts.flattenDiagnosticMessageText(
-            diagnostic.messageText,
-            "\n"
-          );
-
-          this.log(
-            `  ${logSymbols.error} (${line + 1},${character + 1}): ${message}`
-          );
-        } else {
-          this.log(
-            ts.flattenDiagnosticMessageText(diagnostic.messageText, "\n")
-          );
-        }
+    ["errors", "warnings"].forEach((category) => {
```

Comment:
Should we type this a bit better so if we expand this a bit better to include something like `info` we don't have to update it in a ton of places?